src/bots/depositor.py

```diff
@@ -262,3 +247,68 @@ def prepare_and_send_tx(self, module_id: int, quorum: list[DepositMessage]) -> b
         label = 'success' if success else 'failure'
         MODULE_TX_SEND.labels(label, module_id).inc()
         return success
+
+    def _fetch_actual_messages(self) -> list[BotMessage]:
+        # Fetch messages and apply filters
+        actualize_filter = self._get_message_actualize_filter()
+        prefix = self.w3.lido.deposit_security_module.get_attest_message_prefix()
+        sign_filter = get_messages_sign_filter(prefix)
+
+        return self.message_storage.get_messages_and_actualize(lambda x: sign_filter(x) and actualize_filter(x))
+
+    def _get_preferred_to_deposit_modules(self) -> list[int]:
+        # gather quorum
+        now = datetime.now()
+        for module_id in variables.DEPOSIT_MODULES_WHITELIST:
+            if self._get_quorum(module_id):
+                self._module_last_heart_beat[module_id] = now
+
+        # filter out non allow-listed modules
+        module_ids = [
+            module_id
+            for module_id in self.w3.lido.staking_router.get_staking_module_ids()
+            if module_id in variables.DEPOSIT_MODULES_WHITELIST
+        ]
+        # get digests for all the modules
+        module_digests = self.w3.lido.staking_router.get_staking_module_digests(module_ids)
+        # sort modules by validator count
+        sorted_module_digests = sorted(
+            module_digests,
+            key=lambda module_digest: self._validator_difference(module_digest),
+        )
+        # decide if modules are healthy
+        # module[2][0] - module_id
+        modules_healthiness = [(module[2][0], self._is_module_healthy(module[2][0])) for module in sorted_module_digests]
+
+        # take all the modules in sorted order until the first healthy one(including)
+        result = self._take_until_first_healthy_module(modules_healthiness)
+        logger.info({'msg': f'Module iteration order {result}.'})
+        return result
+
+    def _is_module_healthy(self, module_id: int) -> bool:
+        # Check if the quorum cache is valid
+        last_quorum_time = self._module_last_heart_beat[module_id]
+        is_valid_quorum = datetime.now() - last_quorum_time <= timedelta(minutes=variables.QUORUM_RETENTION_MINUTES)
+        logger.info({'msg': f'Is valid quorum {is_valid_quorum}.', 'module_id': module_id})
+
+        # Check if module is available for deposits
+        can_deposit = self.w3.lido.deposit_security_module.can_deposit(module_id)
+        logger.info({'msg': f'Can deposit {can_deposit}.', 'module_id': module_id})
+
+        strategy = self._select_strategy(module_id)
```

Comment:
let's just check that module has one depositable key

Reply:
Done